### PR TITLE
Iox #563 move tests in posh into anonymous namespace

### LINKED ISSUE #563: Move all tests into an anonymous namespace
## Brief feature description

All tests should be self contained in an anonymous namespace.

## Detailed information

Multiple tests files are linked together into one binary. If the tests are not put into an unique namespace, it is possible that multiple helper structs with the same name but different layout are in the same binary. With luck, the linker complains. With bad luck, the address sanitizer detects some weird errors after doing something totally unrelated. Like renaming test files in #561 which lead to a different ordering in the compilation process and made the address sanitizer complain about heap buffer overflow due to 4 tests having a `DummyData` struct and one of them had a different layout.

TODO:
- [x] utils moduletests
- [x] utils integrationtets
- [ ] posh moduletests
- [ ] posh integrationtets
- [ ] dds moduletests
- [ ] binding_c moduletests

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Partially closes #563

### CODE REVIEW

File: iceoryx_posh/test/moduletests/test_popo_trigger.cpp [hunk 1]
```diff
@@ -20,6 +20,8 @@
 #include "test.hpp"
 #include <thread>
 
+namespace
```
Comment (mossmaurice) [inline iceoryx_posh/test/moduletests/test_popo_trigger.cpp:23]: Nitpick : 2020 - 2021

File: iceoryx_posh/test/moduletests/test_popo_trigger_handle.cpp [hunk 2]
```diff
@@ -22,6 +22,8 @@
 #include "test.hpp"
 #include <thread>
 
+namespace
```
Comment (mossmaurice) [inline iceoryx_posh/test/moduletests/test_popo_trigger_handle.cpp:25]: Nitpick : 2020 - 2021

File: iceoryx_posh/test/moduletests/test_roudi_cmd_line_parser.cpp [hunk 3]
```diff
@@ -43,9 +43,7 @@ bool operator==(const CmdLineArgs_t& lhs, const CmdLineArgs_t& rhs)
 } // namespace config
 } // namespace iox
 
-namespace iox
-{
-namespace test
```
Comment (mossmaurice) [inline iceoryx_posh/test/moduletests/test_roudi_cmd_line_parser.cpp:48]: Our current ASPICE SWE.4 examples [states other namespaces](https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/aspice_swe3_4/example/iceoryx_component/test/moduletests/test_example_base_class.cpp#L26-L28). Could you adapt the documentation accordingly?

**Review by mossmaurice — COMMENTED** [review]
LGTM

**Review by mossmaurice — APPROVED** [review]

**Review by mossmaurice — APPROVED** [review]

File: iceoryx_posh/test/integrationtests/test_roudi_findservice.cpp [hunk 4]
```diff
@@ -29,6 +35,15 @@ class RoudiFindService_test : public RouDiServiceDiscoveryTest
     {
     }
 
+    void InitContainer(InstanceContainer& dest, std::vector<std::string> src)
```
[the remark this answers is not in the document]
Comment (elBoberido) [reply iceoryx_posh/test/integrationtests/test_roudi_findservice.cpp:38]: good question. I copied from another file.

[the remark this answers is not in the document]
Comment (elBoberido) [reply iceoryx_posh/test/integrationtests/test_roudi_findservice.cpp]: it seems it was only one place

**Review by mossmaurice — APPROVED** [review]

**Review by elfenpiff — APPROVED** [review]

**Pull request merged**